process/elasticproc/transactions/transactionDBBuilder.go

```diff
@@ -110,19 +112,76 @@ func (dtb *dbTransactionBuilder) prepareTransaction(
 		ReceiverUserName:  []byte(receiverUserName),
 		SenderUserName:    []byte(senderUserName),
 		IsScCall:          isScCall,
-		Operation:         res.Operation,
-		Function:          converters.TruncateFieldIfExceedsMaxLength(res.Function),
 		ESDTValues:        esdtValues,
 		ESDTValuesNum:     esdtValuesNum,
-		Tokens:            converters.TruncateSliceElementsIfExceedsMaxLength(res.Tokens),
 		Receivers:         receiversAddr,
-		ReceiversShardIDs: res.ReceiversShardID,
-		IsRelayed:         res.IsRelayed,
 		Version:           tx.Version,
 		GuardianAddress:   guardianAddress,
 		GuardianSignature: hex.EncodeToString(tx.GuardianSignature),
 		ExecutionOrder:    int(txInfo.ExecutionOrder),
 	}
+
+	isRelayedV3 := len(tx.InnerTransactions) > 0
+	if isRelayedV3 {
+		eTx.Operation = res.Operation
+		dtb.addRelayedV3InfoInIndexerTx(tx, eTx, numOfShards)
+
+		return eTx
+	}
+
+	eTx.Operation = res.Operation
+	eTx.Function = converters.TruncateFieldIfExceedsMaxLength(res.Function)
+	eTx.Tokens = converters.TruncateSliceElementsIfExceedsMaxLength(res.Tokens)
+	eTx.ReceiversShardIDs = res.ReceiversShardID
+	eTx.IsRelayed = res.IsRelayed
+
+	return eTx
+}
+
+func (dtb *dbTransactionBuilder) addRelayedV3InfoInIndexerTx(tx *transaction.Transaction, indexerTx *data.Transaction, numOfShards uint32) {
+	if len(tx.InnerTransactions) == 0 {
+		return
+	}
+
+	innerTxs := make([]*transaction.FrontendTransaction, 0, len(tx.InnerTransactions))
+	receivers := make([]string, 0, len(tx.InnerTransactions))
+	receiversShardIDs := make([]uint32, 0, len(tx.InnerTransactions))
+	for _, innerTx := range tx.InnerTransactions {
+		frontEndTx := &transaction.FrontendTransaction{
+			Nonce:            innerTx.Nonce,
+			Value:            innerTx.Value.String(),
+			Receiver:         dtb.addressPubkeyConverter.SilentEncode(innerTx.RcvAddr, log),
+			Sender:           dtb.addressPubkeyConverter.SilentEncode(innerTx.SndAddr, log),
+			SenderUsername:   innerTx.SndUserName,
+			ReceiverUsername: innerTx.RcvUserName,
+			GasPrice:         innerTx.GasPrice,
+			GasLimit:         innerTx.GasLimit,
+			Data:             innerTx.Data,
+			Signature:        hex.EncodeToString(innerTx.Signature),
+			ChainID:          string(innerTx.ChainID),
+			Version:          innerTx.Version,
+			Options:          innerTx.Options,
+		}
+
+		if len(innerTx.GuardianAddr) > 0 {
```

Comment:
should add guardian addr and guardian signature for the big relayed tx too.. same for non-relayed v3

Reply:
Its added on line 120 and 121

Reply:
missed it